src/libcore/iter.rs

```diff
@@ -149,8 +156,10 @@ pub trait Iterator {
     #[stable(feature = "rust1", since = "1.0.0")]
     fn nth(&mut self, mut n: usize) -> Option<Self::Item> where Self: Sized {
         for x in self.by_ref() {
-            if n == 0 { return Some(x) }
-            n -= 1;
+            n = match n.checked_sub(1) {
+                Some(nn) => nn,
+                None => return Some(x),
+            }
```

Comment:
Yea, it seems you're right, I get similar results with that options. Can't reproduce my results.